### ISSUE TITLE
BUILD fails on kernel-6.12.4-arch1-1

### ISSUE THREAD

**Issue by ManuLinares** [linux-apfs/linux-apfs-rw#87]
Archlinux using this package:https://aur.archlinux.org/packages/linux-apfs-rw-dkms-git

linux-apfs-rw version 0.3.11
linux 6.12.4.arch1-1
dkms 3.1.3-1

<details>
  <summary>make.log</summary>
  

```make
DKMS make.log for linux-apfs-rw/0.3.11 for kernel 6.12.4-arch1-1 (x86_64)
mar 10 dic 2024 14:12:11 -03
make: Entering directory '/usr/lib/modules/6.12.4-arch1-1/build'
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/btree.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/compress.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/dir.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/extents.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/file.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/key.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/libzbitmap.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/lzfse/lzfse_decode.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/lzfse/lzfse_decode_base.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/lzfse/lzfse_fse.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/lzfse/lzvn_decode_base.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/message.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/namei.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/node.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/object.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/snapshot.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/spaceman.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/super.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/symlink.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/transaction.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/unicode.o
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/xattr.o
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c: In function ‘__apfs_write_begin’:
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:568:35: error: passing argument 1 of ‘__block_write_begin’ from incompatible pointer type [-Wincompatible-pointer-types]
  568 |         err = __block_write_begin(page, pos, len, apfs_get_new_block);
      |                                   ^~~~
      |                                   |
      |                                   struct page *
In file included from /var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:7:
./include/linux/buffer_head.h:261:39: note: expected ‘struct folio *’ but argument is of type ‘struct page *’
  261 | int __block_write_begin(struct folio *folio, loff_t pos, unsigned len,
      |                         ~~~~~~~~~~~~~~^~~~~
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c: In function ‘__apfs_write_end’:
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:631:66: error: passing argument 6 of ‘generic_write_end’ from incompatible pointer type [-Wincompatible-pointer-types]
  631 |         ret = generic_write_end(file, mapping, pos, len, copied, page, fsdata);
      |                                                                  ^~~~
      |                                                                  |
      |                                                                  struct page *
./include/linux/buffer_head.h:268:33: note: expected ‘struct folio *’ but argument is of type ‘struct page *’
  268 |                                 struct folio *, void *);
      |                                 ^~~~~~~~~~~~~~
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c: At top level:
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:701:27: error: initialization of ‘int (*)(struct file *, struct address_space *, loff_t,  unsigned int,  struct folio **, void **)’ {aka ‘int (*)(struct file *, struct address_space *, long long int,  unsigned int,  struct folio **, void **)’} from incompatible pointer type ‘int (*)(struct file *, struct address_space *, loff_t,  unsigned int,  struct page **, void **)’ {aka ‘int (*)(struct file *, struct address_space *, long long int,  unsigned int,  struct page **, void **)’} [-Wincompatible-pointer-types]
  701 |         .write_begin    = apfs_write_begin,
      |                           ^~~~~~~~~~~~~~~~
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:701:27: note: (near initialization for ‘apfs_aops.write_begin’)
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:702:27: error: initialization of ‘int (*)(struct file *, struct address_space *, loff_t,  unsigned int,  unsigned int,  struct folio *, void *)’ {aka ‘int (*)(struct file *, struct address_space *, long long int,  unsigned int,  unsigned int,  struct folio *, void *)’} from incompatible pointer type ‘int (*)(struct file *, struct address_space *, loff_t,  unsigned int,  unsigned int,  struct page *, void *)’ {aka ‘int (*)(struct file *, struct address_space *, long long int,  unsigned int,  unsigned int,  struct page *, void *)’} [-Wincompatible-pointer-types]
  702 |         .write_end      = apfs_write_end,
      |                           ^~~~~~~~~~~~~~
/var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.c:702:27: note: (near initialization for ‘apfs_aops.write_end’)
  CC [M]  /var/lib/dkms/linux-apfs-rw/0.3.11/build/xfield.o
make[2]: *** [scripts/Makefile.build:229: /var/lib/dkms/linux-apfs-rw/0.3.11/build/inode.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [/usr/lib/modules/6.12.4-arch1-1/build/Makefile:1936: /var/lib/dkms/linux-apfs-rw/0.3.11/build] Error 2
make: *** [Makefile:224: __sub-make] Error 2
make: Leaving directory '/usr/lib/modules/6.12.4-arch1-1/build'

```

</details>

**Comment by Luflosi** [comment]
This should be fixed in the [v0.3.12 version](https://github.com/linux-apfs/linux-apfs-rw/releases/tag/v0.3.12).

**Comment by ManuLinares** [comment]
Indeed, thanks!
